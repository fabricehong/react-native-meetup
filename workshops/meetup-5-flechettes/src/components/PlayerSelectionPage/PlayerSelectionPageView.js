import React, { Component } from 'react';
import { Text, View, TextInput, Button, Dimensions } from 'react-native';
import PrimaryButton from '../common/PrimaryButton';

const { Twidth, Theight } = Dimensions.get('window');

export default class PlayerSelectionPageView extends Component {
    constructor(props) {
        super(props);
        this.state = { text: 'Player 1', players: ['player'] };
    }

    editPlayer = (text) => {
        let players = [ ...this.state.markers ];
        markers[index] = {...markers[index], key: value};
        this.setState({ markers });
    }

    addPlayer = () => {
        let input = this.state.text.trim();
        if (input && !this.state.players.includes(input)) {
            this.setState({ players :  [...this.state.players, input], text: ''});
        }
    }

    removePlayer (player) {
        this.setState((prevState) => ({ players: prevState.players.filter((ele) => ele !== player) }));
    }

    startGame = () => {
      this.props.navigation.navigate('gamePage')
    };

    render() {
    	const {toutSurUneligne, nomPlayer, lesBoutons, bouton } = styles;
        return (
            <View>
                <Text>Enter your players !</Text>
                <TextInput
                    style={{ height: 40, borderColor: 'gray', borderWidth: 1 }}
                    onChangeText={(text) => this.setState({ text })}
                    value={this.state.text}
                />
                <Button title="add" onPress={this.addPlayer}/>  
                {this.state.players.map((ele) => 
                    <View key={ele} style={toutSurUneligne} >
                        <Text key={ele+'_2'} style={nomPlayer}>Joueur : {ele}</Text>
                        <View key={ele+'_3'} style={lesBoutons}>
				<Button key={ele+'_1'} style={bouton} title="X" onPress={() => this.removePlayer(ele)}/>
				<Button key={ele+'_4'} style={bouton} title="O" onPress={() => this.removePlayer(ele)}/>
			</View>
                    </View> 
                )}
                <PrimaryButton label="Start" onPress={this.startGame}/>
            </View>
        )
    }
}


const styles = {
	toutSurUneligne: {
		flexDirection: 'row',
		borderBottomWidth: 3,
		borderColor: '#d6d7da',
	},
	nomPlayer: {
		flex: 2,
	},
	lesBoutons: {
		flex: 1,
		flexDirection: 'row',
	},
	bouton: {
		width : 20,
		paddingHorizontal: 25, // ne fonctionne pas.		
	},
};



