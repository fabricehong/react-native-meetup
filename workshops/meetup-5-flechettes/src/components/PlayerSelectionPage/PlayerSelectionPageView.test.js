import React from 'react';
import renderer from 'react-test-renderer';
import PlayerSelectionPageView from './PlayerSelectionPageView';

describe('PlayerSelectionPageView', () => {
  test('should renders correctly', () => {
    const tree = renderer.create(<PlayerSelectionPageView/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});