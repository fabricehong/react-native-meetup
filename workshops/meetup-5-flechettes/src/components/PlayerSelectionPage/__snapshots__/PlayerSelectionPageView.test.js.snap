// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerSelectionPageView should renders correctly 1`] = `
<View>
  <Text>
    Enter your players !
  </Text>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    style={
      Object {
        "borderColor": "gray",
        "borderWidth": 1,
        "height": 40,
      }
    }
    underlineColorAndroid="transparent"
    value="Player 1"
  />
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        add
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "borderBottomWidth": 3,
        "borderColor": "#d6d7da",
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "flex": 2,
        }
      }
    >
      Joueur : 
      player
    </Text>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            X
          </Text>
        </View>
      </View>
      <View
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            O
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#aaaaff",
        "margin": 10,
        "padding": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      Start
    </Text>
  </View>
</View>
`;
